stdlib/experimental/geo/geo.flux

```diff
@@ -25,194 +25,200 @@ builtin stLength : (geometry: A, units: {distance: string}) => float where A: Re
-  return
-    tables
-      |> filter(fn: (r) =>
-        if _grid.level == _s2cellIDLevel then
-          contains(value: r.s2_cell_id, set: _grid.set)
-        else
-          contains(value: s2CellIDToken(token: r.s2_cell_id, level: _grid.level), set: _grid.set)
-      )
+    _s2cellIDLevel = if s2cellIDLevel == -1 then
+        tables
+            |> _detectLevel()
+else
+        s2cellIDLevel
+    _grid = getGrid(
+        region: region,
+        minSize: minSize,
+        maxSize: maxSize,
+        level: level,
+        maxLevel: _s2cellIDLevel,
+        units: units,
+    )
+
+    return tables
+        |> filter(
+            fn: (r) => if _grid.level == _s2cellIDLevel then
+                contains(value: r.s2_cell_id, set: _grid.set)
+else
+                contains(value: s2CellIDToken(token: r.s2_cell_id, level: _grid.level), set: _grid.set),
+        )
 }
 
 // Filters records by specified region.
 // It is an exact filter and must be used after `toRows()` because it requires `lat` and `lon` columns in input row sets.
-strictFilter = (tables=<-, region) =>
-  tables
-    |> filter(fn: (r) =>
-      ST_Contains(region: region, geometry: {lat: r.lat, lon: r.lon})
-    )
+strictFilter = (tables=<-, region) => tables
+    |> filter(fn: (r) => ST_Contains(region: region, geometry: {lat: r.lat, lon: r.lon}))
 
 // Two-phase filtering by specified region.
 // Checks to see if data is already pivoted and contains a lat column.
 // Returns pivoted data.
 filterRows = (tables=<-, region, minSize=24, maxSize=-1, level=-1, s2cellIDLevel=-1, strict=true) => {
-  _columns =
-    tables
-      |> columns(column: "_value")
-      |> tableFind(fn: (key) => true )
-      |> getColumn(column: "_value")
-  _rows =
-    if contains(value: "lat", set: _columns) then
-      tables
-        |> gridFilter(region: region, minSize: minSize, maxSize: maxSize, level: level, s2cellIDLevel: s2cellIDLevel)
-    else
-      tables
-        |> gridFilter(region: region, minSize: minSize, maxSize: maxSize, level: level, s2cellIDLevel: s2cellIDLevel)
-        |> toRows()
-  _result =
-    if strict then
-      _rows
-        |> strictFilter(region)
-    else
-      _rows
-  return _result
+    _columns = tables
+        |> columns(column: "_value")
+        |> tableFind(fn: (key) => true)
+        |> getColumn(column: "_value")
+    _rows = if contains(value: "lat", set: _columns) then
+        tables
+            |> gridFilter(
+                region: region,
+                minSize: minSize,
+                maxSize: maxSize,
+                level: level,
+                s2cellIDLevel: s2cellIDLevel,
+            )
+else
+        tables
+            |> gridFilter(
+                region: region,
+                minSize: minSize,
+                maxSize: maxSize,
+                level: level,
+                s2cellIDLevel: s2cellIDLevel,
+            )
+            |> toRows()
```

Comment:
I'd like to understand better how we decide whether or not to parenthesize the `alternate` of a conditional. will this interpret the `|> toRows()` as being part of the `alternate`, or will `|> toRows()` be interpreted as following both branches of if conditional? I guess it interprets it as the former because `|>` has higher precedence than `if/then/else`. Do I have that right?